app/models/student_type_scholarship.rb

```diff
@@ -3,9 +3,14 @@ class StudentTypeScholarship < ApplicationRecord
   belongs_to :type_scholarship
 
   before_validation :set_date
+  before_update :set_date
   
   validates :date, presence: true
 
+  delegate :description, to: :type_scholarship, prefix: true
```

Comment:
Esto se puede unir en una sola linea:
`delegate :description, :scholarship, to: :type_scholarship, prefix: true`